giraffe/src/utils/newTable.ts

```diff
@@ -69,8 +70,19 @@ class SimpleTable implements Table {
     return column.type
   }
 
+  getOriginalColumnType(columnKey: string): FluxDataType {
+    const column = this.columns[columnKey]
+
+    if (!column) {
+      return null
+    }
+
+    return column.originalType
+  }
+
   addColumn(
     columnKey: string,
+    originalType: FluxDataType,
```

Comment:
i wonder if we can overload this function to keep backwards compatibility with the existing interface. idk how to do it in typescript, but in plain js it's something like:
```
addColumn(...args) {
   if (args.length === 4) {
         const [ columnKey, type, data, name ] = args
         // try and make something up for the original type
   } else if (args.length === 5) {
         const [ columnKey, originalType, type, data, name ] = args
        // continue like normal
  }
}
```